pkg/app/api/grpcapi/piped_api.go

```diff
@@ -149,7 +149,19 @@ func (a *PipedAPI) ReportPipedMeta(ctx context.Context, req *pipedservice.Report
 			return nil, status.Error(codes.Internal, "failed to update the piped metadata")
 		}
 	}
-	return &pipedservice.ReportPipedMetaResponse{}, nil
+
+	var piped *model.Piped
+	piped, err = a.pipedStore.GetPiped(ctx, pipedID)
```

Comment:
You can use this helper to find the piped. It is also handling the error for us.

``` golang
	piped, err := getPiped(ctx, a.pipedStore, pipedID, a.logger)
        if err != nil {
		return nil, err
	}
```

Reply:
902afd4